### ISSUE TITLE
Two test.hs files

### ISSUE THREAD

**Issue by arianvp** [rudymatela/leancheck#7]
There are two test.hs files.  (Test.hs and test.hs) This totally wreacks havoc on both windows and Darwin as they are case insensitive filesystems. Making it impossible to compile the tests

**Comment by rudymatela** [comment]
@arianvp I have reported these [issues upstream to Microsoft Corporation](https://youtu.be/dQw4w9WgXcQ) and [Apple Inc.](https://youtu.be/dQw4w9WgXcQ) so that they fix their broken filesystems by adding case sensitivity.  ... ... Just kidding.

Thank you very much for reporting this issue.  I can't believe for how long that stood there undetected by me.

Commit f07481599528e2b6809e61804e20d1d1e9f1594b fixes the issue by renaming `test.hs` to `test-main.hs`.  I have also added an automated test (f07481599528e2b6809e61804e20d1d1e9f1594b) to assure I don't forget about case-insensitive filesystems in the future.

[As soon as CI passes](https://travis-ci.org/rudymatela/leancheck/builds/432682636), I'll upload a new version on Hackage.  Then, I'll let you know here.
